libvast/vast/plugin.hpp

```diff
@@ -100,17 +100,59 @@ class plugin {
   [[nodiscard]] virtual const char* name() const = 0;
 };
 
+// -- component plugin --------------------------------------------------------
+
+/// A base class for plugins that spawn components in the NODE.
+/// @relates plugin
+class component_plugin : public virtual plugin {
+public:
+  /// Creates an actor as a component in the NODE.
+  /// @param node A pointer to the NODE actor handle.
+  /// @returns The actor handle to the NODE component.
+  virtual system::component_plugin_actor
+  make_component(system::node_actor::pointer node) const = 0;
+};
+
 // -- analyzer plugin ----------------------------------------------------------
 
 /// A base class for plugins that hook into the input stream.
-/// @relates plugin
-class analyzer_plugin : public virtual plugin {
+/// @relates component_plugin
+class analyzer_plugin : public virtual component_plugin {
 public:
+  /// Gets the ANALYZER actor spawned by the plugin, if it is still running.
+  inline system::analyzer_plugin_actor
+  get_analyzer(system::node_actor::pointer node) const {
```

Comment:
I'm starting to see a lot of `get_*`. Our coding style does not encourage this. What speaks against using just `analyzer(node)` here?

Reply:
Fine by me. Will rename.